File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.py
def check(cs):
    for _ in range(len(cs)):
        cs = cs.replace('[]', '')
        cs = cs.replace('()', '')
        cs = cs.replace('{}', '')
        
    return True if not len(cs) else False

def solution(s):
    answer = 0

    for i in range(len(s)):
        if check(s):
            answer += 1
        s = s[1:] + s[0]

    return answer

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.swift
import Foundation

func check(_ s: String) -> Bool {
    var stack = [Character]()
    let brackets: [Character: Character] = [")": "(", "}": "{", "]": "["]
    
    for char in s {
        if let bracket = brackets[char] {
            if let top = stack.last, top == bracket {
                stack.removeLast()
            } else {
                return false
            }
        } else {
            stack.append(char)
        }
    }
    return stack.isEmpty
}

func solution(_ s:String) -> Int {
    var answer = 0
    var tmp = s
    for _ in 0..<s.count {
        if check(tmp) {
            answer += 1
        }
        tmp.append(tmp.removeFirst())
    }
    return answer
}